src/App.jsx

```diff
@@ -55,8 +58,12 @@ function App() {
   };
 
   useEffect(() => {
-    retrieveGeoData(searchInput);
-  }, [searchInput]);
+    let cityToSearch;
+
+    if (loggedIn && searchInput === '') cityToSearch = currentUser.homeCity;
+    else cityToSearch = searchInput;
```

Comment:
This works as intended, there's nothing wrong. However I'm just confused how it works and the logic. If the searchInput value is Miami, and the user logs in, I would assume the if statement doesn't trigger yet it does. I may be missing something.

Reply:
That's definitely a reasonable thing to assume. I had to clear `searchInput` on a new login using `onLoginToggle()` on line 87:
```
              onLoginToggle={(login) => {
                setSearchInput('');
                setLoggedIn(login);
              }}
```

Reply:
Okay I see how that works. I saw that you set it to an empty string but I assumed it was just to reset the searchInput for another reasons. There was definitely a lot of moving parts so I missed that. Thank you for clarifying!